### PR TITLE
fix: Accordion design kit alignment (#1512) 

### PR DESCRIPTION
**Pull request by castastrophe**
## Align pfe-accordion with the kit

- [x] Fix typography on disclosure

### #1512 
- [x] use chevron icon from PF Core
- [x] ensure chevron aligned in disclosure variant
- [x] correct chevron direction
- [x] focus state
- [x] hover/active state background color for header region

### #1403
- [x] Allow badge in the accordion component

### Testing instructions

- [x] *Component design matches mock-ups #877*
- [x] *Click toggles the accordion state*
- [x] *Space or enter toggles the accordion state*

#### Browser requirements

Your component should work in all of the following environments:

- [x] Latest 2 versions of Edge
- [x] Internet Explorer 11 (should be useable, not pixel perfect)
- [x] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [x] Firefox 78 (or latest version for Red Hat Enterprise Linux distribution)
- [x] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [x] Latest 2 versions of Safari
- [x] Android mobile device (such as the Galaxy S9)
- [x] Apple mobile device (such as the iPhone X)
- [x] Apple tablet device (such as the iPhone Pro)

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one request or issue (no stragglers or scope-creep).
- [x] Browser testing passed.
- [x] Changelog updated.
- [x] Documentation (README.md, WHY.md, etc.) updated or added.
- [x] Approved by designer.

### Merging

Please **squash** when merging and ensure your commit message uses [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) formatting.

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: elements/pfelement/src/pfelement.js [hunk 1]
```diff
@@ -356,7 +356,6 @@ class PFElement extends HTMLElement {
     super();
 
     this._pfeClass = pfeClass;
```
Comment (castastrophe) [inline elements/pfelement/src/pfelement.js:358]:
```suggestion
    this._pfeClass = pfeClass;
    this.tag = pfeClass.tag;
```

File: elements/pfelement/src/pfelement.js [hunk 2]
```diff
@@ -475,7 +475,10 @@ class PFElement extends HTMLElement {
    */
   render() {
     this.shadowRoot.innerHTML = "";
-    this.template.innerHTML = this.html;
+    this.template.innerHTML = "";
+
+    if (this.styles) this.template.innerHTML += this.styles;
```
Comment (castastrophe) [inline elements/pfelement/src/pfelement.js:480]: Separating out styles and HTML makes it easier to put the template in the JS file but keep the stylesheet in Sass

File: elements/pfelement/src/pfelement.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { autoReveal } from "./reveal.js";
 import { isAllowedType, isValidDefaultType } from "./attrDefValidators.js";
 
-// Import polyfills: Array.includes, Object.entries, String.startsWith, Element.closest, Element.matches, Array.prototype.find
+// Import polyfills: Array.includes, Object.entries, String.startsWith, Element.closest, Element.matches, Array.prototype.find, Array.prototype.findIndex
```
Comment (castastrophe) [inline elements/pfelement/src/pfelement.js:4]: Moved the findIndex polyfill to pfelement and was able to remove several polyfill files

**Comment by coreyvickery** [comment]
@castastrophe Final feedback.

> Website reviewed in Chrome 92.0.4515.107

## Accordion

Thin white line

* Seems to be a very thin white line at the top when I hover over the panels below the first one (first one is fine)
* For the text, I changed the weight to `500` and the text is where I want it to be/aligns with the kit
* Something I'd like to discuss with another dev is `pfe-icon` about available icons and how to customize them if you wanted to create a separate issue for that (or I can)

## Disclosure

Disclosure padding

* The two Disclosure examples on the website have different heights/padding, let's shoot for the same height, padding, and the caret vertically-aligned to the text and vertically-aligned to the center of the component
* Seems there's a slight delay before the caret rotates (like 0.5 seconds), the caret rotating and the panel expand animation should happen at the same time
* NEW tag should be to the right of the text, let's try 24px of padding

Everything else looks great!

**Review by heyMP — APPROVED** [review]

File: elements/pfe-sass/mixins/_components.scss [hunk 4]
```diff
@@ -221,9 +220,9 @@
   width:  $size;
   transition: transform 0.15s;
   @if $state == open {
-    border-width: .1em .1em 0 0;
+    border-width: $width $width 0 0;
     border-bottom: 0;
-    transform: rotate(-45deg);
+    transform: rotate(135deg);
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply elements/pfe-sass/mixins/_components.scss:225]: This is addressing Corey's feedback that the chevrons were pointing in the wrong direction before.

**Comment by castastrophe** [comment]
@coreyvickery

> For the text, I changed the weight to 500 and the text is where I want it to be/aligns with the kit

We are not using variable fonts yet in production so even though you will see a difference on your system when you adjust the weight (because you likely have the font installed on your OS), users only have access to a set number of weights which are the following:

```
  // These font weights are specific to the Red Hat font families from Google Fonts:
  font-weight--light:        300,
  font-weight--normal:       400,
  font-weight--semi-bold:    600,
  font-weight--bold:         700,
```

Let me know if you prefer normal or semi-bold.  Thanks!

**Comment by castastrophe** [comment]
- [x] Thin white line at the top fixed
- [x] The caret vertically-aligned to the text and vertically-aligned to the center of the component
- [x] Fixed delay before the caret rotates
- [x] NEW tag should be to the right of the text, let's try 24px of padding

> Something I'd like to discuss with another dev is pfe-icon about available icons and how to customize them if you wanted to create a separate issue for that (or I can)

Please do create an issue and assign it to Potter for discussion!

> The two Disclosure examples on the website have different heights/padding, let's shoot for the same height, padding, and the caret vertically-aligned to the text and vertically-aligned to the center of the component

I just want to note that the accent region is a slot so we don't have a lot of control over what goes in there or what styles they use.  If someone puts an element in there that is larger than 1em, we're going to see some variance (that's what you're seeing with the disclosure that has the pfe-badge element in it).  I've tidied up alignment a bit to try to make this less obvious.  Let me know if that works.

**Comment by coreyvickery** [comment]
@castastrophe 

> Let me know if you prefer normal or semi-bold. Thanks!

`semi-bold` please.

> I've tidied up alignment a bit to try to make this less obvious. Let me know if that works.

It does, thanks.

**Pull request merged**